packages/lib/src/interactions/SelectToZoom.tsx

```diff
@@ -1,69 +1,76 @@
 
 function SelectToZoom(props: Props) {
-  const { canvasSize, canvasRatio, visRatio, dataToWorld } =
+  const { canvasSize, canvasRatio, visRatio, worldToData } =
     useVisCanvasContext();
 
   const { width, height } = canvasSize;
   const keepRatio = visRatio !== undefined;
 
   const camera = useThree((state) => state.camera);
-  const moveCameraTo = useMoveCameraTo();
+  const zoomOnBox = useZoomOnBox();
 
-  function onSelectionEnd(selection: Selection) {
-    const [worldStart, worldEnd] = getRatioRespectingRectangle(
-      ...selection.data,
-      keepRatio ? canvasRatio : undefined
-    ).map(dataToWorld); // work in world coordinates as we need to act on the world camera
+  function computeZoomBox(worldSelection: Rect3): Box {
+    const { position, scale } = camera;
+    const zoomBox = Box.fromPoints(...worldSelection);
 
-    if (worldStart.x === worldEnd.x || worldStart.y === worldEnd.y) {
-      return;
+    if (!keepRatio) {
+      return zoomBox;
     }
 
-    const zoomRect = getEnclosedRectangle(worldStart, worldEnd);
-    const { center: zoomRectCenter } = zoomRect;
+    const fovBox = Box.empty(position).expandBySize(
+      width * scale.x,
+      height * scale.y
+    );
+
+    return zoomBox.expandToRatio(canvasRatio).keepWithin(fovBox);
+  }
 
-    // Change scale first so that moveCameraTo computes the updated camera bounds
-    camera.scale.set(zoomRect.width / width, zoomRect.height / height, 1);
-    camera.updateMatrixWorld();
+  function onSelectionEnd(selection: Selection) {
+    const [worldStart, worldEnd] = selection.world;
+    if (worldStart.x === worldEnd.x || worldStart.y === worldEnd.y) {
+      return;
+    }
 
-    moveCameraTo(zoomRectCenter);
+    zoomOnBox(computeZoomBox(selection.world));
   }
 
   return (
     <SelectionTool id="SelectToZoom" onSelectionEnd={onSelectionEnd} {...props}>
-      {({ data: [dataStart, dataEnd] }) => (
-        <>
-          <SelectionRect
-            startPoint={dataStart}
-            endPoint={dataEnd}
-            fill="white"
-            stroke="black"
-            fillOpacity={keepRatio ? 0 : 0.25}
-            strokeDasharray={keepRatio ? '4' : undefined}
-          />
-          {keepRatio && (
-            <RatioSelectionRect
+      {({ data: [dataStart, dataEnd], world: worldSelection }) => {
+        const zoomBox = computeZoomBox(worldSelection);
+
+        return (
+          <>
+            <SelectionRect
               startPoint={dataStart}
               endPoint={dataEnd}
-              ratio={canvasRatio}
-              fillOpacity={0.25}
               fill="white"
               stroke="black"
+              fillOpacity={keepRatio ? 0 : 0.25}
+              strokeDasharray={keepRatio ? '4' : undefined}
             />
-          )}
-        </>
-      )}
+            {keepRatio && (
+              <SelectionRect
+                startPoint={worldToData(zoomBox.min)}
+                endPoint={worldToData(zoomBox.max)}
```

Comment:
If we end up adding low-level components that accept `worldCoords: Rect3`, we might not need the extra `zoomBox` constant and `worldToData` conversions.